langchain/chains/llm_router_chain/base.py

```diff
@@ -0,0 +1,103 @@
+import re
+from types import FunctionType
+from typing import Dict, List, Any
+from pydantic import Extra, root_validator
+
+from langchain import LLMChain, BasePromptTemplate
+from langchain.chains.base import Chain
+from langchain.chains.conversation.prompt import PROMPT
+from langchain.input import get_color_mapping
+from langchain.memory import ConversationBufferWindowMemory
+from langchain.schema import BaseMemory
+
+
+class RouterChain(LLMChain):
+    """
+    Router chain that picks the most relevant model to call based on vector queries.
+    The chain includes support for memory
+    """
+
+    memory: BaseMemory = ConversationBufferWindowMemory(k=1)
+    fuzzy_match_threshold = 1.5
+    """Default memory store."""
+    prompt: BasePromptTemplate = PROMPT
+    """Default conversation prompt to use."""
+    last_chain: Chain = None
+    chains: Dict[str, Chain]
+    strip_outputs: bool = False
+    input_key: str = "input"  #: :meta private:
+    output_key: str = "output"  #: :meta private:
+    vector_lookup_fn: FunctionType = None
```

Comment:
better typing here

if this is implemented using Vectorstores, can we use that?

Reply:
I left this completely up to the caller. Did not want to enforce a vector store (although my initial change did have that). I feel for some cases even keeping those in memory is a possibility which does not require a heavy vector store integration.